ckanext/harvest/harvesters/ckanharvester.py

```diff
@@ -374,7 +366,7 @@ def import_stage(self,harvest_object):
             # Find any extras whose values are not strings and try to convert
             # them to strings, as non-string extras are not allowed anymore in
             # CKAN 2.0.
-            for key in package_dict['extras'].keys():
+            for key in package_dict.get('extras', {}).keys():
```

Comment:
`extras` is a list of dicts, so calling `.keys()` won't work. We need to iterate of them and check the `key` value.
The logic on the lines below is also broken for v3 dicts